### PR TITLE
Pirate Uplink and New Drip.

### PR DESCRIPTION
**Pull request by Salvantrix**
Adds in a fully operational pirate coconut uplink, as well as new drippy drip.

Listed as draft as awaiting updated icon sprites.

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

- add: New Pirate Drip and Coconut uplink!
<!--

### CODE REVIEW

**Comment by Leander-0** [comment]
I do think the prices should be all around lower, the pirate revolver is much worse than the regular one for example and there is no way for pirates to gain more doubloons.

**Comment by Leander-0** [comment]
Note that the pirate hardsuits are in a good price range but they are really weak right now, the hardsuit should be buffed to the gold-silver regular NFSD suits and on a different PR add a new more powerful pirate hardsuit.

File: Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        - PirateCaptain
+
+- type: listing
+  id: UplinkPirateBlunderbuss
+  name:  uplink-pirate-blunderbuss-name
+  description: uplink-pirate-blunderbuss-desc
+  productEntity: WeaponShotgunBlunderbuss
+  icon: { sprite: Objects/Weapons/Guns/Shotguns/blunderbuss.rsi, state: icon }
+  cost:
+    Doubloon: 4
```
Comment (Leander-0) [inline Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml:145]: Should be chaged to 1 - 2, the blunderbuss only loads a single shotgun slug and needs to be wielded to shoot.

File: Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+          - PirateUplink
+
+- type: listing
+  id: UplinkPirateRevolver
+  name:  uplink-pirate-revolver-name
+  description: uplink-pirate-revolver-desc
+  productEntity: WeaponRevolverPirate
+  icon: { sprite: Objects/Weapons/Guns/Revolvers/pirate_revolver.rsi, state: icon }
+  cost:
+    Doubloon: 3
```
Comment (Leander-0) [inline Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml:161]: also change to 1 since only has 5 shots and fires slower than a regular revolver.

**Comment by Leander-0** [comment]
The individual cannon ammo should be in bundles of 5 for 2 DB, instead of individual ones for 1 DB, since the all cannon bundles. but that can be done on a different PR to add a box or small crate for those to come in.

File: Resources/Textures/_NF/Clothing/Head/Hats/pirate_hat.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    "version": 1,
+    "license": "CC-BY-NC-SA-4.0",
```
Comment (whatston3) [inline Resources/Textures/_NF/Clothing/Head/Hats/pirate_hat.rsi/meta.json:3]: Licenses are a bit all over the place.  Any reason why?

File: Resources/Prototypes/_NF/Entities/Clothing/Neck/cloaks.yml [hunk 4]
```diff
@@ -16,4 +16,13 @@
...
   - type: Sprite
-    sprite: _NF/Clothing/Neck/Cloaks/sr.rsi
\ No newline at end of file
+    sprite: _NF/Clothing/Neck/Cloaks/sr.rsi
+
+- type: entity
+  parent: ClothingNeckBase
+  id: ClothingNeckCloakPirateParrot
+  name: polly the pirate parrot
+  description: A mute parrot named polly, that enjoys crackers and cannons.
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/Neck/cloaks.yml:25]: Proper nouns.  Also, could give a tip that it's actually equippable.
```suggestion
  name: Polly the pirate parrot
  description: A mute parrot named Polly. Loves sitting on shoulders, crackers, and cannons.
```

File: Resources/Locale/en-US/_NF/store/uplink-catalog.ftl [hunk 5]
```diff
@@ -141,4 +141,39 @@ uplink-security-stingergrenade-box-desc = A box containing 4 stinger grenades.
...
+uplink-pirate-cannon-name = Pirate Cannon
+uplink-pirate-cannon-desc = Kaboom!
+uplink-pirate-cannonball-name = Cannonball
+uplink-pirate-cannonball-desc = A ball made to be shot from a cannon.
+uplink-pirate-grapeshot-name = Grapeshot
+uplink-pirate-grapeshot-desc = A cluster ball made to shoot a wide range of small projectiles.
+uplink-pirate-glassshot-name = Glassshot
+uplink-pirate-glassshot-desc = A brittle glass ball that will fire multiple projectiles in a tight spread.
+uplink-pirate-crateammo-name = Cannonball Crate
+uplink-pirate-crateammo-desc = A crate filled with cannonballs.
```
Comment (whatston3) [inline Resources/Locale/en-US/_NF/store/uplink-catalog.ftl:177]: It's a chest, call it a chest.
```suggestion
uplink-pirate-crateammo-name = Cannonball Chest
uplink-pirate-crateammo-desc = A chest filled with cannonballs.
```

File: Resources/Locale/en-US/_NF/store/uplink-catalog.ftl [hunk 6]
```diff
@@ -141,4 +141,39 @@ uplink-security-stingergrenade-box-desc = A box containing 4 stinger grenades.
...
+store-category-pirateexplosives = Explosives
+
+uplink-pirate-hardsuit-name = Pirate HardSuit
+uplink-pirate-hardsuit-desc = A heavy space suit that provides some basic protection from the cold harsh realities of deep space.
+uplink-pirate-hardsuit-captain-name = Pirate Captain's Hardsuit
+uplink-pirate-hardsuit-captain-desc = An ancient armored hardsuit, perfect for defending against space scurvy and toolbox-wielding scallywags.
+uplink-pirate-crate-captain-name = Pirate Captain's Crate
+uplink-pirate-crate-captain-desc = A crate filled with the necessary goodies for a pirate captain.
+uplink-pirate-crate-name = Pirate Crate
+uplink-pirate-crate-desc = A crate filled with the necessary goodies for a pirate.
```
Comment (whatston3) [inline Resources/Locale/en-US/_NF/store/uplink-catalog.ftl:159]: Replace "crate" with chest - they're chests.
```suggestion
uplink-pirate-crate-captain-name = Pirate Captain's Chest
uplink-pirate-crate-captain-desc = A chest filled with the necessary goodies for a pirate captain.
uplink-pirate-crate-name = Pirate Chest
uplink-pirate-crate-desc = A chest filled with the necessary goodies for a pirate.
```

**Review by whatston3 — CHANGES_REQUESTED** [review]
I like this feature a lot.  Everything looks good.  Left a few suggestions for your consideration (mostly text), and I have a branch with a blinky pirate uplink.  The balance changes recommended by Leander seem reasonable to me.

Sorry for dragging my heels on this.

**Comment by Salvantrix** [comment]
> I like this feature a lot.  Everything looks good.  Left a few suggestions for your consideration (mostly text), and I have a branch with a blinky pirate uplink.  The balance changes recommended by Leander seem reasonable to me.
> 
> Sorry for dragging my heels on this.

Oops to a couple of those, I needed to do a pass as some of that is just laziness on my part. I'll get that updated tonight.

File: Resources/Prototypes/_NF/Store/presets.yml [hunk 7]
```diff
@@ -27,3 +27,15 @@
...
+
+- type: storePreset
+  id: StorePresetPirateUplink
+  storeName: Pirate Uplink
+  categories:
+  - UplinkPirateHardsuits
+  - UplinkPirateWeapons
+  - UplinkPirateAmmo
+  - UplinkPirateUtility
+  - UplinkPirateExplosives
```
[the remark this answers is not in the document]
Comment (Salvantrix) [reply Resources/Prototypes/_NF/Store/presets.yml:39]: Not sure to be honest, I used the exact template from the security uplink

File: Resources/Textures/_NF/Clothing/Head/Hats/pirate_hat.rsi/meta.json (hunk 3 above)
Comment (Salvantrix) [reply Resources/Textures/_NF/Clothing/Head/Hats/pirate_hat.rsi/meta.json:3]: Updating now to fix

File: Resources/Prototypes/_NF/Entities/Objects/Specific/pirate.yml [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+# Uplinks
+- type: entity
+  parent: BaseItem
+  id: BasePirateUplink
+  name: pirate uplink
+  description: A coconut with an antenna?
+  suffix: Empty
+  noSpawn: true
+  components:
+  - type: Sprite
```
[the remark this answers is not in the document]
Comment (Salvantrix) [reply Resources/Prototypes/_NF/Entities/Objects/Specific/pirate.yml:58]: This looks much better! Is this something already being PR'd or can it just be smacked down in here before merge?

[the remark this answers is not in the document]
Comment (whatston3) [reply Resources/Prototypes/_NF/Entities/Objects/Specific/pirate.yml:58]: Not under PR, but feel free to take the commit (linked below).  If you'd like, I can add directly into this branch.  Do you prefer the smaller or larger version?

https://github.com/whatston3/frontier-station-14/commit/770cdecef2d974dd8d9b30aad440c81a0f371164

File: Resources/Prototypes/_NF/Store/presets.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (whatston3) [reply Resources/Prototypes/_NF/Store/presets.yml:39]: Weird.  Thanks!

**Review by whatston3 — COMMENTED** [review]
Looking good, pardon the nitpicks - let me know when your final changes are up and I'll give it another read.

File: Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+          - PirateUplink
+    - !type:BuyerJobCondition
+      whitelist:
+        - PirateCaptain
+
+- type: listing
+  id: UplinkPirateShotgunAmmo
+  name:  uplink-pirate-shotgunammo-name
+  description: uplink-pirate-shotgunammo-desc
+  productEntity: BoxMagazineShotgun
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml:212]: Is this supposed to be the shotgun slug dispenser?  This is not the shotgun slug dispenser.  I can change it on my end, but seems odd to get a bunch of bulldog mags.

File: Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+  icon: { sprite: Objects/Storage/boxes.rsi, state: shellslug }
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/pirate_uplink_catalog.yml:213]: #1426 fixes this.

**Review by whatston3 — APPROVED** [review]
Happy with the head of the branch as it is.  Requesting another sanity check, but my testing looked good.

**Comment by Salvantrix** [comment]
Fixes make this work as intended on my end, ready for a triple check if needed.

**Comment by whatston3** [comment]
Merge conflicts should be resolved.

**Pull request merged**